### PR TITLE
storage: Improve doc comments delete methods

### PR DESCRIPTION
**Pull request by ifraixedes**
What: Improve the doc comments of storage delete methods adding a description of its behavior. Also remove some `TODO` comments which don't apply anymore.

Why: They help to not have to inspect the logic to know their behavior, for example, I needed to know about it when working on https://github.com/storj/storj/pull/3590

Please describe the tests: N/A
Please describe the performance impact: N/A

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: storage/filestore/store.go [hunk 1]
```diff
@@ -87,7 +87,10 @@ func (store *blobStore) StatWithStorageFormat(ctx context.Context, ref storage.B
 	return info, Error.Wrap(err)
 }
 
-// Delete deletes blobs with the specified ref
+// Delete deletes blobs with the specified ref.
+//
+// It doesn't return an error if the blog isn't found by any reason or it cannot
```
Comment (mniewrzal) [inline storage/filestore/store.go]: typo `blog`

File: storagenode/pieces/store.go [hunk 2]
```diff
@@ -253,6 +253,9 @@ func (store *Store) ReaderWithStorageFormat(ctx context.Context, satellite storj
 }
 
 // Delete deletes the specified piece.
+//
+// It doesn't return an error if the blog isn't found by any reason or it cannot
```
Comment (mniewrzal) [inline storagenode/pieces/store.go]: typo `blog`

File: storage/filestore/dir.go [hunk 3]
```diff
@@ -368,12 +368,23 @@ func (dir *Dir) iterateStorageFormatVersions(ctx context.Context, ref storage.Bl
 }
 
 // Delete deletes blobs with the specified ref (in all supported storage formats).
+//
+// It doesn't return an error if the blob is not found by any reason or it
```
Comment (mobyvb) [inline storage/filestore/dir.go]:
```suggestion
// It doesn't return an error if the blob is not found for any reason or it
```

File: storage/filestore/dir.go [hunk 4]
```diff
@@ -368,12 +368,23 @@ func (dir *Dir) iterateStorageFormatVersions(ctx context.Context, ref storage.Bl
...
-// DeleteWithStorageFormat deletes the blob with the specified ref for one specific format version
+// DeleteWithStorageFormat deletes the blob with the specified ref for one
+// specific format version. The method tries the following strategies, in order
+// of preference until one succeeds:
+//
+// * moves the blob to garbage dir.
+// * directly deletes the blob.
+// * push the blobs to queue for retrying later.
+//
+// It doesn't return an error if the piece isn't found by any reason.
```
Comment (mobyvb) [inline storage/filestore/dir.go]:
```suggestion
// It doesn't return an error if the piece isn't found for any reason.
```

File: storage/filestore/store.go [hunk 5]
```diff
@@ -87,7 +87,10 @@ func (store *blobStore) StatWithStorageFormat(ctx context.Context, ref storage.B
 	return info, Error.Wrap(err)
 }
 
-// Delete deletes blobs with the specified ref
+// Delete deletes blobs with the specified ref.
+//
+// It doesn't return an error if the blob isn't found by any reason or it cannot
```
Comment (mobyvb) [inline storage/filestore/store.go]:
```suggestion
// It doesn't return an error if the blob isn't found for any reason or it cannot
```

**Pull request merged**